x-pack/plugins/apm/server/lib/metrics/by_agent/java/thread_count/index.ts

```diff
@@ -43,18 +43,21 @@ const chartBase: ChartBase = {
 
 export async function getThreadCountChart({
   environment,
+  kuery,
   setup,
   serviceName,
   serviceNodeName,
 }: {
   environment?: string;
+  kuery?: string;
```

Comment:
Should these be optional always? Can we avoid it?

Reply:
Can't think of a case where it _wouldn't_ be optional, since you can always have an empty kuery bar and it's present on most pages.

Reply:
if it's empty I'd say it's still a string? But making it optional makes it easy to accidentally leave out

Reply:
I'd like to revisit the "requiredness" of parameters shortly in a future PR. Since `useUrlParams` always returns an object with everything optional, it's a bit tricky to make something required, and there are some endpoints where kuery and environment aren't there at all. If we can make some improvements to the client-side routes to have more type information about their query parameters we can make this less error-prone.

Reply:
I think we can just make kuery required in the UrlParams type. I can have a look if you want (and also make the uiFilters type change).

Reply:
Please do, but this PR is already changing 144 files so I don't want to make it too much bigger.

Reply:
@smith Alright, if you can create the issue I'll take a stab on a follow-up PR, that way you're not blocked by me.

Reply:
Quid pro quo 😬

Reply:
#92286

Reply:
Pleasure doing business 🤝